### ISSUE TITLE
add arbitrary metadata tags to sentences

### ISSUE THREAD

**Issue by jonorthwash** [jonorthwash/ud-annotatrix#16]
Come up with some way, both in interface and annotation of the various formats, to tag sentences with arbitrary metadata tags.  Examples of useful tags might include `checked_by_Gary`, `ignore`, `check_again`, etc.

**Comment by maryszmary** [comment]
Should the metadata tags be stored in the comments before the sentence or somewhere else?

**Comment by jonorthwash** [comment]
Probably in comments somewhere, yes.  And I think before the sentence makes sense for most formats too.

**Comment by ftyers** [comment]
My suggestion would be to have e.g.

```
# sent =  (7)
# text = abaj-ni urši kung b-ʡald-aq-ib aʔib
# gloss = mother.OBL-ERG boy.ABL book.ABL N-hide.PFV-CAUS-AOR drive.PFV-AOR
# text[eng] = ‘Mother made her son hide a book.’
# tags = gloss_error, check, missing_punctuation
1	Абайни	абай	NOUN	_	Case=Erg|Number=Sing	_	_	_Gloss=mother.OBL-ERG|Transcript=abaj-ni
2	урши	урши	NOUN	_	Case=Abs|Number=Sing	_	_	_Gloss=boy.ABL|Transcript=urši
3	кунг	кунг	NOUN	_	Case=Abs|Number=Sing	_	_	_Gloss=book.ABL|Transcript=kung
4	бгӏалдахъиб	_	_	_	Aspect=Perf|Gender=Neut|Tense=Aor	_	_	_	Gloss=N-hide.PFV-CAUS-AOR|Transcript=b-ʡald-aq-ib
5	аъиб	_	_	_	Aspect=Perf|Tense=Aor	_	_	_Gloss=drive.PFV-AOR|Transcript=aʔib
```

**Comment by ftyers** [comment]
Comments don't apply to `sdparse`, and for CG, we can just do the same as in CoNLL-U (e.g. lines beginning with # are comments).

**Comment by jonorthwash** [comment]
Right.  And okay :)

**Comment by ftyers** [comment]
A couple more questions:

1) Should the list of tags work like in Github with the labels, where you have to define them before, or should it work more like in the GSOC interface where you type them and they complete, but new ones are added automatically ? 

2) How should the interface layout work ? Should it go to one side, above or below? What would be most convenient ?

**Comment by jonorthwash** [comment]
1. Probably like the GSoC interface, but we don't have to decide this ahead of time, I think.
2. Doesn't matter—this is a design choice that can be decided as design happens.  Should be made to be easily customisable with css.  Also, a lot of the interface is going to change anyway, so it's premature to make decisions like this now.  And we should have keyboard shortcuts to access it too.

**Comment by ftyers** [comment]
In the Breton treebank I've been using:
```
# labels = foo bar baz_bur
```
I prefer this to `tags`

**Comment by jonorthwash** [comment]
The only thing different is the name, though, right?

**Comment by ftyers** [comment]
This has been (re)implemented in 9c859ff, but lots of functionality is missing, should there be separate issues for the different things we want to do ? Or should we just list them here?

**Comment by jonorthwash** [comment]
I'm happy either way.

**Comment by ftyers** [comment]
Ok, let's close this for now.
